adapters/smilewanted/smilewanted.go

```diff
@@ -0,0 +1,162 @@
+package smilewanted
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type SmileWantedAdapter struct {
+	URI string
+}
+
+func (a *SmileWantedAdapter) MakeRequests(requestIn *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+
+	request := *requestIn
+
+	if len(request.Imp) == 0 {
+		return nil, []error{&errortypes.BadInput{
+			Message: "No impression in the bid request",
+		}}
+	}
+
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(request.Imp[0].Ext, &bidderExt); err != nil {
+		return nil, []error{&errortypes.BadInput{
+			Message: "ext.bidder not provided",
+		}}
+	}
+
+	var smilewantedExt openrtb_ext.ExtImpSmileWanted
+	if err := json.Unmarshal(bidderExt.Bidder, &smilewantedExt); err != nil {
+		return nil, []error{&errortypes.BadInput{
+			Message: "ext.bidder.zoneId not provided",
+		}}
+	}
+
+	if smilewantedExt.ZoneID == "" {
+		return nil, []error{&errortypes.BadInput{
+			Message: "ZoneId is empty",
+		}}
+	}
```

Comment:
I see you added the min length constraint to the json schema. That now allows you to safely remove the following code from your adapter, since it is just checking validity of the bidder extension which is already performed by PBS-Core.

```
var bidderExt adapters.ExtImpBidder
if err := json.Unmarshal(request.Imp[0].Ext, &bidderExt); err != nil {
    return nil, []error{&errortypes.BadInput{
        Message: "ext.bidder not provided",
    }}
}

var smilewantedExt openrtb_ext.ExtImpSmileWanted
if err := json.Unmarshal(bidderExt.Bidder, &smilewantedExt); err != nil {
    return nil, []error{&errortypes.BadInput{
        Message: "ext.bidder.zoneId not provided",
    }}
}
```